### PR TITLE
docs: update build docs,support Powershell on Windows

### PR DESCRIPTION
**Pull request by nashaofu**
#### Description of Change
update the build docs, support the Windows command line and PowerShell

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/main/docs/development/testing.md)
- [x] relevant documentation, tutorials, templates and examples are changed or added
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Updated windows build docs

### CODE REVIEW

File: docs/development/build-instructions-gn.md [hunk 1]
```diff
@@ -110,20 +110,49 @@ $ export CHROMIUM_BUILDTOOLS_PATH=`pwd`/buildtools
 On Windows:
 
 ```sh
+# CMD
 $ cd src
 $ set CHROMIUM_BUILDTOOLS_PATH=%cd%\buildtools
+
+# Powershell
+$ cd src
+$ $env:CHROMIUM_BUILDTOOLS_PATH = "$(Get-Location)\buildtools"
```
Comment (erickzhao) [inline docs/development/build-instructions-gn.md:119]: nit: can we correct the capitalization from `CMD` to `cmd` and `Powershell` to `PowerShell`

```suggestion
# cmd
$ cd src
$ set CHROMIUM_BUILDTOOLS_PATH=%cd%\buildtools

# PowerShell
$ cd src
$ $env:CHROMIUM_BUILDTOOLS_PATH = "$(Get-Location)\buildtools"
```

File: docs/development/build-instructions-gn.md [hunk 2]
```diff
@@ -110,20 +110,49 @@ $ export CHROMIUM_BUILDTOOLS_PATH=`pwd`/buildtools
...
+```
+
+On Windows:
+
 ```sh
+# CMD
 $ gn gen out/Release --args="import(\"//electron/build/args/release.gn\")"
+
+# Powershell
+$ gn gen out/Release --args="import(\`"//electron/build/args/release.gn\`")"
```
Comment (erickzhao) [inline docs/development/build-instructions-gn.md:155]: question: is the `\` before the quote still necessary in PowerShell after we add the backtick escape?

File: docs/development/build-instructions-gn.md (hunk 1 above)
Comment (nashaofu) [reply docs/development/build-instructions-gn.md]: fixed

File: docs/development/build-instructions-gn.md (hunk 2 above)
Comment (nashaofu) [reply docs/development/build-instructions-gn.md:155]: Yes, It is necessary, otherwise it will report an error

**Review by erickzhao — APPROVED** [review]

**Review by erickzhao — APPROVED** [review]
docs: lgtm

**Pull request merged**